src/renderer.js

```diff
@@ -110,7 +114,22 @@ Renderer.prototype.resize = function (resolution) {
   return false;
 };
 
-function updateUniforms(gl, camera, uniforms) {
+Renderer.prototype.texturize = function (image) {
+  if (!image) {
+    this.texture = null;
+    this.glTexture = createTexture(this.gl, block);
+    return Promise.resolve();
+  }
+
+  return getImage(image)
+    .then(function (pixels) {
+      this.texture = image;
+      this.glTexture = createTexture(this.gl, pixels);
+    }.bind(this))
+    .catch(function (err) { console.log(err); });
+};
```

Comment:
I'm not super sure if the old textures are automatically released or not. There is an explicit deleteTexture call that you may want to introduce in the future (not necessary now). If the textures are not automatically freed, switching back and forth between textures will leak memory.